### PR TITLE
fix: release v0.37.2 runtime signature backend dependencies

### PR DESCRIPTION
**Pull request by djm81**
# Description

Release `v0.37.2` to fix runtime module-signature verification failures after installation.

This PR includes:
- hard runtime dependencies for signature verification backend (`cryptography`, `cffi`)
- module installer handling for unavailable signature backend (explicit behavior in strict/non-strict paths)
- test updates for signature-backend-unavailable scenarios
- version alignment to `0.37.2` and changelog update

**Fixes** #N/A

**New Features** #N/A

**Contract References**: No new public contract decorators were added in this change.

## Type of Change

Please check all that apply:

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔒 Contract enforcement (adding/updating `@icontract` decorators)
- [x] 🧪 Test enhancement (scenario tests, property-based tests)
- [ ] 🔧 Refactoring (code improvement without functionality change)

## Contract-First Testing Evidence

**Required for all changes affecting CLI commands or public APIs:**

### Contract Validation

- [ ] **Runtime contracts added/updated** (`@icontract` decorators on public APIs)
- [ ] **Type checking enforced** (`@beartype` decorators applied)
- [ ] **CrossHair exploration** completed: `hatch run contract-test-exploration`
- [ ] **Contract violations** reviewed and addressed

### Test Execution

- [ ] **Contract validation**: `hatch run contract-test-contracts` ✅
- [ ] **Contract exploration**: `hatch run contract-test-exploration` ✅
- [ ] **Scenario tests**: `hatch run contract-test-scenarios` ✅
- [ ] **Full test suite**: `hatch run contract-test-full` ✅

### Test Quality

- [x] **CLI commands tested** with typer test client
- [ ] **Edge cases covered** with Hypothesis property tests
- [x] **Error handling tested** with invalid inputs
- [x] **Rich console output verified** manually or with snapshots

## How Has This Been Tested?

**Contract-First Approach**: Focused regression tests and type-checking were run for changed paths.

### Manual Testing

- [x] Tested CLI commands manually
- [x] Verified rich console output
- [x] Tested with different input scenarios
- [x] Checked error messages for clarity

### Automated Testing

- [ ] Contract validation passes
- [ ] Property-based tests cover edge cases
- [ ] Scenario tests cover user workflows
- [ ] All existing tests still pass

### Test Environment

- Python version: 3.12
- OS: Ubuntu

## Checklist

- [x] My code follows the style guidelines (PEP 8, ruff format, isort)
- [x] I have performed a self-review of my code
- [ ] I have added/updated contracts (`@icontract`, `@beartype`)
- [ ] I have added/updated docstrings (Google style)
- [x] I have made corresponding changes to documentation
- [ ] My changes generate no new warnings (basedpyright, ruff, pylint)
- [ ] All tests pass locally
- [x] I have added tests that prove my fix/feature works
- [x] Any dependent changes have been merged

## Quality Gates Status

- [x] **Type checking** ✅ (`hatch run type-check`)
- [ ] **Linting** ✅ (`hatch run lint`)
- [ ] **Contract validation** ✅ (`hatch run contract-test-contracts`)
- [ ] **Contract exploration** ✅ (`hatch run contract-test-exploration`)
- [ ] **Scenario tests** ✅ (`hatch run contract-test-scenarios`)

## Screenshots/Recordings (if applicable)

N/A

### CODE REVIEW

File: src/specfact_cli/registry/module_installer.py [hunk 1]
```diff
@@ -337,6 +350,17 @@ def verify_module_artifact(
...
+            if _is_signature_backend_unavailable(exc):
+                if require_signature and not allow_unsigned:
+                    logger.warning("Module %s: signature is required but backend is unavailable", meta.name)
+                    return False
+                if not _SIGNATURE_BACKEND_WARNING_EMITTED:
+                    logger.warning(
+                        "Module signature backend unavailable (%s). Falling back to checksum-only verification.",
+                        exc,
+                    )
+                    _SIGNATURE_BACKEND_WARNING_EMITTED = True
```
[the remark this answers is not in the document]
Comment (djm81) [reply src/specfact_cli/registry/module_installer.py]: fix applied
